README.md

```diff
@@ -180,6 +182,8 @@ type LiveStreamMethods = {
   startStreaming: (streamKey: string, url?: string) => void;
   // Stops the stream
   stopStreaming: () => void;
+  // Sets the zoom ratio. Can be used if you want to control the zooming.
+  setZoomRatio: (zoomRatio: number) => void;
```

Comment:
Use property instead?

Reply:
My idea was, normally with anything that is a gesture or slider. You will 99% of the time have access to a `onValueChange` Which in this case should be called continuously while the value is changing. 

Do you mean component prop? Or ref property?

As a component prop wouldnt work, because it can cause a rerender. The rerenders would pile up and eventually crash the app from what i have seen.
Ill play a little with it to see what i can do.

Reply:
ref property such as `isMuted`

Reply:
What do you say if we have both? Both can be used in different cases, and the performance vary.
If you have a list of available zooms that you can switch between, it will not be a problem to pass it down as a prop because it wont be updated as often.
But if you want to achieve a smooth zoom with a slider, then the setZoomRatio will be a better choice simply because it wont rerender the react part.
Ive set up a demo of this in the example.

Updating a zoomRatio prop while sliding a slider using a state, will cause the zooming to be a little laggy, maybe only every 5-10 tick will actually be passed down to the zoom.
I would really like to see how it would perform with an animated value, but havent been able to convert that to a number.
Using the method setZoomRatio with a slider is fairly smooth. It is slightly laggy compared to the native zoom, but still barely noticeable.

What do you think?

Reply:
I am not an expert of RN.
I guess you have much more experiences than I do 👍 
So I am okay with implementing both ways. 
The only that triggers me is that there should be a way to get current zoom ratio (which is possible with the zoomRatio prop).

Reply:
Alright, ill take a look at getting current zoomRatio

Reply:
So i have now gone it over.
I am not entirely sure what you mean by getting the current zoom by the regular prop?

If i want to get the current zoomRatio from the native side. How would i do that with the prop?
Normally i would create a 
```
@ReactMethod
  fun getZoomRatio(view: ReactNativeLiveStreamView, promise: Promise) {
    try {
      val _zoomRatio: Float = view.zoomRatio
      promise.resolve(_zoomRatio)
    } catch (e: java.lang.Exception) {
      promise.reject(e)
    }
  }
```
in a native module which gets that value.
Is there a better way?

Reply:
Oups, sorry. I haven't answered here.

Is there a way get get the zoom value with the `zoomRatio` prop like doing thing like this:
`const zoomRatio = ref.current?.zoomRatio`
I guess no that is why we don't understand each other ;)
In this case, the `getZoomRatio` would be perfect!

If `setZoomRatio` has better performance than `zoomRatio`, could you stash pop`setZoomRatio`?

Reply:
With setZoomRatio you would be able to do some cool stuff with RN gestures & animations. Which is not possible with the prop.

Last week i went through everything, and i couldnt find a way to do it neatly. - Wanted to get zoomRange, current zoom from the native side to js.
The way i used to do it, was to have the streamer variable in a singleton. That way you can interact with it through a module.
This way is viable. But i didnt have enough time to maintain this on top of everything else. So eventually it stopped working.
This specific version was forked from some turkish guy who deleted it from github.
https://github.com/BlueBazze/rtmp/blob/master/android/src/main/java/com/reactnativertmppublisher/RTMPModule.java

Reply:
Ok.
If I have understand correctly, the slider should use `setZoomRatio` instead of the `zoomRatio` props. Right?

For a getter on zoom ratio, unfortunately I don't have time to have a look on it this week. It would be nice to have but let say we drop it for now till someone asks for it.

Reply:
Ill quickly adjust the example.

I agree. - The zoomRatio will always start at 1. And people can keep their own state when they setZoomRatio for now.